modules/platforms/cpp/network/src/network/length_prefix_codec.cpp

```diff
@@ -45,7 +45,7 @@ namespace ignite
 
         DataBuffer LengthPrefixCodec::Decode(DataBuffer& data)
         {
-            if (packet.IsValid() && packet.Get()->Length() == (PACKET_HEADER_SIZE + packetSize))
+            if (packet.IsValid() && packetSize != -1 && packet.Get()->Length() == (PACKET_HEADER_SIZE + packetSize))
```

Comment:
Probably just this nitpick. Joining these two branches with an if-else and putting the 2nd case first would be easier to follow.